cfme/containers/provider/openshift.py

```diff
@@ -20,37 +20,31 @@ class OpenshiftProvider(ContainersProvider):
     db_types = ["Openshift::ContainerManager"]
 
     def __init__(self, name=None, credentials=None, key=None, zone=None, hostname=None, port=None,
-                 sec_protocol=None, hawkular_sec_protocol=None, provider_data=None, appliance=None):
+                 sec_protocol=None, hawkular_sec_protocol=None, provider_data=None, appliance=None,
+                 hawkular_hostname=None, hawkular_api_port=None):
         super(OpenshiftProvider, self).__init__(
             name=name, credentials=credentials, key=key, zone=zone, hostname=hostname, port=port,
             sec_protocol=sec_protocol, hawkular_sec_protocol=hawkular_sec_protocol,
-            provider_data=provider_data, appliance=appliance)
+            provider_data=provider_data, appliance=appliance,
+            hawkular_hostname=hawkular_hostname, hawkular_api_port=hawkular_api_port)
 
     def href(self):
         return self.appliance.rest_api.collections.providers\
             .find_by(name=self.name).resources[0].href
 
-    def _form_mapping(self, create=None, hawkular=False, **kwargs):
-        if self.appliance.version > '5.8.0.3' and hawkular:
-            sec_protocol = kwargs.get('sec_protocol'),
-            hawkular_hostname = kwargs.get('hostname')
-            hawkular_sec_protocol = kwargs.get('hawkular_sec_protocol')
-        elif self.appliance.version > '5.8.0.3' and not hawkular:
+    def _form_mapping(self, create=None, **kwargs):
+        if self.appliance.version > '5.8.0.3':
             sec_protocol = kwargs.get('sec_protocol')
-            hawkular_hostname = None
-            hawkular_sec_protocol = None
         else:
             sec_protocol = None
-            hawkular_hostname = None
-            hawkular_sec_protocol = None
         return {'name_text': kwargs.get('name'),
                 'type_select': create and 'OpenShift',
                 'hostname_text': kwargs.get('hostname'),
                 'port_text': kwargs.get('port'),
                 'sec_protocol': sec_protocol,
                 'zone_select': kwargs.get('zone'),
-                'hawkular_hostname': hawkular_hostname,
-                'hawkular_sec_protocol': hawkular_sec_protocol}
+                'hawkular_hostname': kwargs.get('hostname'),
```

Comment:
it doesn't matter. _form_mapping is called in provider's create method. It takes value from provider class. If you passed hawkular_hostname to provider class in from_config method, I suppose it has to be used here instead of hostname. Have I missed something ?